pd/src/state/writer.rs

```diff
@@ -238,6 +238,36 @@ impl Writer {
         .execute(&mut dbtx)
         .await?;
 
+        // Drop quarantined notes associated with a validator slashed in this block
```

Comment:
Here is where we apply the changes resulting from slashing: by dropping the notes associated with a validator which are in quarantine, and removing the nullifiers _both_ from quarantine and from the **main nullifier set** which makes them spendable again.